nixos/modules/services/monitoring/prometheus/exporters/dcgm.nix

```diff
@@ -0,0 +1,175 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+}:
+with lib; let
+  cfg = config.services.prometheus.exporters.dcgm;
+in {
+  port = 9400;
+  extraOpts = {
+    collectors = mkOption {
+      type = types.path;
+      default = "${pkgs.prometheus-dcgm-exporter}/share/etc/dcgm-exporter/default-counters.csv";
+      defaultText = literalExpression ''
+        ''${pkgs.prometheus-dcgm-exporter}/share/etc/dcgm-exporter/default-counters.csv
+      '';
+      example = literalExpression ''
+        ''${pkgs.prometheus-dcgm-exporter}/share/etc/dcgm-exporter/1.x-compatability-metrics.csv
+      '';
+      description = lib.mdDoc ''
+        Path to file containing DCGM fields to collect.
+      '';
+    };
+    address = mkOption {
+      type = types.str;
+      default = "localhost";
+      description = lib.mdDoc ''
+        Address of listening http server.
+      '';
+    };
+    collectInterval = mkOption {
+      type = types.ints.unsigned;
+      default = 30000;
+      description = lib.mdDoc ''
+        Interval of time at which point metrics are collected. Unit is milliseconds.
+      '';
+    };
+    kubernetes = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Enable kubernetes mapping metrics to kubernetes pods.
+      '';
+    };
+    useOldNamespace = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Use old 1.x namespace.
+      '';
+    };
+    configmapData = mkOption {
+      type = types.str;
+      default = "none";
+      description = lib.mdDoc ''
+        ConfigMap namespace and name containing DCGM fields to collect.
+      '';
+    };
+    remoteHostengineInfo = mkOption {
+      type = types.str;
+      default = "${config.services.dcgm.bindInterface}:${toString config.services.dcgm.port}";
```

Comment:
I think this breaks when `bindInterface == "ALL"`. Is there a way we can make it have a sane value in that case?

Reply:
I feel like supporting `ALL` in the first place is a little weird, but we could map it to `0.0.0.0`. I believe it should be functionally the same

Reply:
Maybe we could map the original `ALL` to `0.0.0.0`? I think it's also fine to map this to `127.0.0.1` if `ALL`.

On another note, I think this should be hidden behind an `mkIf config.services.dcgm.enable`